routes/dashboard/_middleware.ts

```diff
@@ -26,18 +26,10 @@ export async function handler(
 
     const { data: { session } } = await supabaseClient.auth.getSession();
     assert(session);
-    ctx.state.session = session;
-
-    const { data } = await supabaseClient
-      .from("customers")
-      .select("stripe_customer_id, is_subscribed")
-      .single()
-      .throwOnError();
 
-    ctx.state.subscription = {
-      stripeCustomerId: data!.stripe_customer_id,
-      isSubscribed: data!.is_subscribed,
-    };
+    ctx.state.session = session;
+    ctx.state.supabaseClient = supabaseClient;
+    ctx.state.customer = await createOrGetCustomer(supabaseClient, stripe);
```

Comment:
This would make a DB request every time the middleware is executed, no? 

I'd say we shouldn't call `createOrGetCustomer` in the middleware, but only actually call it when we need to access the stripe customer id. Wdyt?

Reply:
Good point. I've turned it into a helper to be called whenever needed now in #28.